rclc/CMakeLists.txt

```diff
@@ -4,8 +4,28 @@ project(rclc)
 
 find_package(ament_cmake REQUIRED)
 find_package(rcl REQUIRED)
+find_package(rosidl_generator_c REQUIRED)
 
 ament_export_include_directories(include)
+include_directories(include)
+
+set(rclc_sources
+  "src/functions.c"
+)
+set_source_files_properties(${rclc_sources} PROPERTIES LANGUAGE "C")
```

Comment:
Is this necessary? (I would expect CMake to figure out that `.c` files are c language)
If it's necessary, could we define the CMake project as a C project rather than explicitly setting the source files properties (e.g. https://github.com/ros2/rclpy/blob/74b5f1801e497bad17ab81713bba645e343aeff4/rclpy/CMakeLists.txt#L3)?